hazelcast-client-new/src/test/java/com/hazelcast/client/quorum/cache/ClientCacheReadQuorumTest.java

```diff
@@ -0,0 +1,190 @@
+    private static ICache<Integer, String> cache1;
+    private static ICache<Integer, String> cache2;
+    private static ICache<Integer, String> cache3;
+    private static ICache<Integer, String> cache4;
+    private static ICache<Integer, String> cache5;
+
+    private static TestHazelcastFactory factory;
+
+    @BeforeClass
+    public static void initialize()
+            throws Exception {
+        QuorumConfig quorumConfig = new QuorumConfig();
+        quorumConfig.setName(QUORUM_ID);
+        quorumConfig.setType(QuorumType.READ);
+        quorumConfig.setEnabled(true);
+        quorumConfig.setSize(3);
+
+        CacheSimpleConfig cacheConfig = new CacheSimpleConfig();
+        cacheConfig.setName(CACHE_NAME_PREFIX + "*");
+        cacheConfig.setQuorumName(QUORUM_ID);
+        factory = new TestHazelcastFactory();
+        cluster = new PartitionedCluster(factory).createFiveMemberCluster(cacheConfig, quorumConfig);
+        initializeClients();
+        initializeCaches();
+        cluster.splitFiveMembersThreeAndTwo();
+    }
+
+    private static void initializeClients() {
+        c1 = factory.newHazelcastClient(getClientConfig(cluster.h1));
+        c2 = factory.newHazelcastClient(getClientConfig(cluster.h2));
+        c3 = factory.newHazelcastClient(getClientConfig(cluster.h3));
+        c4 = factory.newHazelcastClient(getClientConfig(cluster.h4));
+        c5 = factory.newHazelcastClient(getClientConfig(cluster.h5));
+    }
+
+    private static void initializeCaches() {
+        cachingProvider1 = HazelcastClientCachingProvider.createCachingProvider(c1);
+        cachingProvider2 = HazelcastClientCachingProvider.createCachingProvider(c2);
+        cachingProvider3 = HazelcastClientCachingProvider.createCachingProvider(c3);
+        cachingProvider4 = HazelcastClientCachingProvider.createCachingProvider(c4);
+        cachingProvider5 = HazelcastClientCachingProvider.createCachingProvider(c5);
+
+        final String cacheName = CACHE_NAME_PREFIX + randomString();
+        cache1 = (ICache)cachingProvider1.getCacheManager().getCache(cacheName);
+        cache2 = (ICache)cachingProvider2.getCacheManager().getCache(cacheName);
+        cache3 = (ICache)cachingProvider3.getCacheManager().getCache(cacheName);
+        cache4 = (ICache)cachingProvider4.getCacheManager().getCache(cacheName);
+        cache5 = (ICache)cachingProvider5.getCacheManager().getCache(cacheName);
+    }
+
+    private static ClientConfig getClientConfig(HazelcastInstance instance) {
+        ClientConfig clientConfig = new ClientConfig();
+        Address address = getNode(instance).address;
+        clientConfig.getNetworkConfig().addAddress(address.getHost() + ":" + address.getPort());
+        clientConfig.getGroupConfig().setName(instance.getConfig().getGroupConfig().getName());
+        return clientConfig;
+    }
+
+    @AfterClass
+    public static void killAllHazelcastInstances() throws IOException {
+        HazelcastInstanceFactory.terminateAll();
```

Comment:
I think you need to call `terminateAll()` on `factory` instance otherwise clients won't be closed correctly.